From patchwork Tue Dec  4 11:24:51 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot] tools: add a generic config for native tools building
X-Patchwork-Submitter: Otavio Salvador <otavio@ossystems.com.br>
X-Patchwork-Id: 1007581
Message-Id: <20181204112452.10544-1-otavio@ossystems.com.br>
To: U-Boot Mailing List <u-boot@lists.denx.de>
Cc: Alexander Graf <agraf@suse.de>, Ryder Lee <ryder.lee@mediatek.com>,
 Otavio Salvador <otavio@ossystems.com.br>,
 Vagrant Cascadian <vagrant@debian.org>,
 Heinrich Schuchardt <xypron.glpk@gmx.de>,
 Alexey Brodkin <alexey.brodkin@synopsys.com>,
 Michal Simek <michal.simek@xilinx.com>
Date: Tue,  4 Dec 2018 09:24:51 -0200
From: Otavio Salvador <otavio@ossystems.com.br>
List-Id: U-Boot discussion <u-boot.lists.denx.de>

The motivation for this is to allow distributions to distribute all
possible tools in a generic way, avoiding the need of specific tools
building for each machine.

Especially on OpenEmbedded / Yocto Project ecosystem, it is very
common each BSP to end providing their specific tools when they need
to generate images for some SoC (e.g MX23 / MX28 in meta-freescale
case).

Using this, we can package the tools doing:

$: make tools-all_defconfig
$: make tools-all

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---

 configs/tools-all_defconfig | 3 +++
 tools/Makefile              | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)
 create mode 100644 configs/tools-all_defconfig

diff --git a/configs/tools-all_defconfig b/configs/tools-all_defconfig
new file mode 100644
index 0000000000..be69bdf43a
--- /dev/null
+++ b/configs/tools-all_defconfig
@@ -0,0 +1,3 @@
+CONFIG_SYS_TEXT_BASE=0
+CONFIG_FIT=y
+CONFIG_FIT_SIGNATURE=y
diff --git a/tools/Makefile b/tools/Makefile
index c93d17a42f..29ff6ed221 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -125,7 +125,7 @@ fit_info-objs   := $(dumpimage-mkimage-objs) fit_info.o
 fit_check_sign-objs   := $(dumpimage-mkimage-objs) fit_check_sign.o
 file2include-objs := file2include.o
 
-ifneq ($(CONFIG_MX23)$(CONFIG_MX28),)
+ifneq ($(CONFIG_MX23)$(CONFIG_MX28)$(CONFIG_FIT_SIGNATURE),)
 # Add CONFIG_MXS into host CFLAGS, so we can check whether or not register
 # the mxsimage support within tools/mxsimage.c .
 HOSTCFLAGS_mxsimage.o += -DCONFIG_MXS
